node.go

```diff
@@ -71,16 +71,28 @@ func (n *node) sendPayments() {
 			if owed > 0 {
 				p, err := n.m.MakePayment(owed, c)
 				if err != nil {
-					log.Printf("Failed to make a payment to %d : %v", c, err)
+					log.Printf("Failed to make a payment to %s : %v", c, err)
 					break
 				}
-				n.Router.SendPayment(p)
+				n.Router.RecordPayment(Payment{n.id.PublicKey().Hash(), c, owed})
+				n.Router.SendPaymentHash(c, p)
 			}
 		}
 		n.l.Unlock()
 	}
 }
 
+func (n *node) receivePayments() {
+	for h := range n.Router.PaymentHashes() {
```

Comment:
It's implemented in payment.go, specificallly the payment handler inside the router.